index.ts

```diff
@@ -123,6 +123,18 @@ export default function mem<
 			maxAge: maxAge ? Date.now() + maxAge : Number.POSITIVE_INFINITY,
 		});
 
+		if (typeof maxAge === 'number' && maxAge !== Number.POSITIVE_INFINITY) {
+			const timer = setTimeout(() => {
+				cache.delete(key);
+			}, maxAge);
+
+			timer.unref?.();
+
+			const timers = cacheTimerStore.get(fn) ?? new Set();
+			timers.add(timer as any); // eslint-disable-line @typescript-eslint/no-unsafe-argument
```

Comment:
Are you looking for this? 📎 

```suggestion
			const timers = cacheTimerStore.get(fn) ?? new Set<NodeJS.Timeout>();
			timers.add(timer);
```

Reply:
This package is environment independent, so I didn't want to use any Node.js specific types.

Reply:
I don’t think this type makes it into `distribution`?

Reply:
Correct. It's technically fine, just my preference not to use it as it.

Reply:
Can the same code work with Set<number>? The WeakMap type above uses it. This would let you move the assertion to `setTimeout() as number` instead of using any and eslint-disable

Reply:
```suggestion
			const timers = cacheTimerStore.get(fn) ?? new Set<number>();
			timers.add(timer as number);
```